FreeRTOS_IP.c

```diff
@@ -2025,6 +2026,18 @@ static eFrameProcessingResult_t prvAllowIPPacket( const IPPacket_t * const pxIPP
                  * broadcast address. */
                 eReturn = eReleaseBuffer;
             }
+            else if( memcmp( ( void * ) &xBroadcastMACAddress,
+                             ( void * ) &( pxIPPacket->xEthernetHeader.xSourceAddress ),
+                             sizeof( MACAddress_t ) ) == 0 )
+            {
+                /* Ethernet source is a broadcast address. */
+                eReturn = eReleaseBuffer;
+            }
+            else if( ( ipFIRST_MULTI_CAST_IPv4 <= ulHostEndianSrcIPAddr ) && ( ulHostEndianSrcIPAddr < ipLAST_MULTI_CAST_IPv4 ) )
```

Comment:
What about this change:
~~~c
- else if( ( ipFIRST_MULTI_CAST_IPv4 <= ulHostEndianSrcIPAddr ) && ( ulHostEndianSrcIPAddr < ipLAST_MULTI_CAST_IPv4 ) )
+ else if( xIsIPv4Multicast( ulSourceIPAddress ) == pdTRUE )
~~~
and you won't need the help variable `ulHostEndianSrcIPAddr` anymore.

Reply:
Done.